code/modules/library/skill_books.dm

```diff
@@ -0,0 +1,617 @@
+			if(2) // basic > adept
+				name = "intermediate [skill_name] textbook"
+				title = pick("Training Yourself in [skill_name]: Volume [rand(1,10)]","Level 2 [skill_name]")
+				desc = "A copy of [title] by [author]. Dry and long, but not unmanageable. Basic knowledge is required to understand the concepts written."
+			if(3) // adept > expert
+				name = "advanced [skill_name] textbook"
+				title = pick("Getting the Experience You Need with [skill_name]", "Level 3 [skill_name]")
+				desc = "A copy of [title] by [author]. Those not already trained in the subject will have a hard time reading this. Try not to drop it either, it will put a hole in the floor."
+			if(4) //expert > prof
+				name = "theoretical [skill_name] textbook"
+				title = pick("Treatise on Theoretical [skill_name]","Level 4 [skill_name]", "Master [skill_name] in [rand(100,999)] easy steps!")
+				desc = "A copy of [title] by [author]. Significant experience in the subject is required to read this incredibly information dense block of paper. Sadly, does not come in audio form."
+	. = ..()
+
+/datum/skill_buff/skill_book
+	limit = 1
+
+/obj/item/book/skill/attack_self(mob/user)
+	if(!ez_read &&!user.skill_check(SKILL_LITERACY, SKILL_BASIC))
+		to_chat(user, SPAN_WARNING(pick("Haha, you know you can't read. Good joke. Put [title] back.","You open up [title], but there aren't any pictures, so you close it again.","You don't know how to read! What good is this [name] to you?!")))
+		return
+
+	if(reading) // close book, get rid of buffs
+		src.unlearn(user)
+		to_chat(user, SPAN_NOTICE("You shut [title]. That's enough learning for now."))
+		reading = FALSE
+		return
+
+	if(user.too_many_buffs(/datum/skill_buff/skill_book))
+		to_chat(user, SPAN_WARNING("You can't read two books at once!"))
+		return
+
+	if(!user.skill_check(skill, skill_req))
+		to_chat(user, SPAN_WARNING("[title] is too advanced for you! Try something easier, perhaps the \"For Idiots\" edition?"))
+		return
+	if(user.get_skill_value(skill) > skill_req)
+		to_chat(user, SPAN_WARNING("You already know everything [title] has to teach you!"))
+		return
+	
+	
+	to_chat(user, SPAN_NOTICE("You open the [title] and start reading..."))
+	if(user.do_skilled(4 SECONDS, SKILL_LITERACY, src, 0.75))
+		var/list/buff = list()
+		buff[skill] = 1
+		user.buff_skill(buff, buff_type = /datum/skill_buff/skill_book)
+		reading = TRUE
+		to_chat(user, SPAN_NOTICE("You find the information you need! Better keep the page open to reference it."))
+	else
+		to_chat(user, SPAN_WARNING("You lose your place on the page."))
+		return
+
+// buff removal
+/obj/item/book/skill/proc/unlearn(var/mob/user)
+	var/list/F = user.fetch_buffs_of_type(/datum/skill_buff/skill_book, 0)
+	for(var/datum/skill_buff/skill_book/S in F)
+		S.remove()
+
+// Remove buffs when book goes away
+/obj/item/book/skill/dropped(mob/user)
+	if(reading)
+		to_chat(user, SPAN_WARNING("You lose the [name] and the page you were on!"))
```

Comment:
`SPAN_WARNING` is usually for failure messages generally - would use `SPAN_DANGER` for this and for the failure message above.